### DISCUSSION TITLE
Keychain access upon app startup

### DISCUSSION THREAD

**Discussion by sarensw** [supabase/supabase#28132] [Questions]
Hi all,

I have included the Supabase Swift Client into my macOS app without any problems. Development went smooth. Now I have archived the app, notarized it and exported it so that I can distribute the app through the marketplace and the website. 

When I started to test the exported app I noticed that upon start of the app the user is requested to grant access to the local keychain. The message says

> AppName wants to use your confidential information stored in "supabase.gotrue.swift" in your keychain

<img width="452" alt="image" src="https://github.com/user-attachments/assets/8e95e0cc-738a-46fb-9581-80be7f24a448">

It only happens on my dev machine. On my old MacBook, which I still use to test the app on an older macOS version, this request does not come up.

I don't want user to see this request. Can someone please shed some light on this? Why does it pop up on my dev machine, and why not on the other machine? Will it pop up for the app users? Especially if they use supabase for development as well?

Regards, Stephan

**Comment by LiuLawrence45** [comment]
Did you hear back? How did you solve this?

**Reply by sarensw** [reply]
I did not resolve this yet. I have removed supabase for the time being from the app.

**Reply by LiuLawrence45** [reply]
Interesting... I'm facing the exact same issue right now.

**Reply by sarensw** [reply]
The feature that supabase would have enabled for my app is not of high priority. That's why for now, it is ok. But I'd still love to hear if you will be able to solve this issue.

**Reply by LiuLawrence45** [reply]
It's a huge deal for me... we're using Supabase for auth. Have spent basically the entire day doing auth for Swift. Not very good support it seems.

**Reply by LiuLawrence45** [reply]
Just spent the past 4-5 hours trying to figure it out. Looks like no options are working. 

Referenced previous threads (here: https://github.com/supabase/supabase-swift/issues/516). I tried the following

1. Downgrading to 1.14 and removing all keychain capabilities (didn't work)
2. Upgrading back to 2.24, but still didn't work (with custom storage option with KeyChain group)
3. Tried a few other permutations of these all (1.14 with keychain, 1.14 with keychain + groups, etc...)

Overall, though, I'm only seeing the issue pop up whenever I notarize my app. I don't have the annoying prompt pop up at all whenever the app is not notarized though. 

The only option I see now is just implementing a custom storage solution for the time being. It's not secure compared to PKCE flow but not sure what else to do, since using supabase for auth (Microsoft + Google) :(

**Comment by elrid** [comment]
Here's how you need to define your supabase client to fix this issue:
```
let supabase = SupabaseClient(
    supabaseURL: URL(string: AuthConfig.supabaseURL)!,
    supabaseKey: AuthConfig.supabaseAnonKey,
    options: SupabaseClientOptions(
        auth: SupabaseClientOptions.AuthOptions(
            storage: KeychainLocalStorage(service: "com.myapp.bundle")
        )
    )
)
```

Instead of "com.myapp.bundle" use your bundle id, or the keychain you use.
